coreneuron/io/reports/report_configuration_parser.cpp

```diff
@@ -59,62 +61,49 @@ enum class TargetType
 /*
  * Split filter string ("mech.var_name") into mech_id and var_name
  */
-void parse_filter_string(char* filter, ReportConfiguration& config) {
-    char* token = strtok(filter, ".");
-    if (!token) {
-        std::cerr << "Error : Invalid report variable, should be mch_name.var_name" << std::endl;
-        nrn_abort(1);
-    }
-    strcpy(config.mech_name, token);
-    token = strtok(nullptr, "\n");
-    if (!token) {
-        std::cerr << "Error : Invalid report variable, should be mch_name.var_name" << std::endl;
-        nrn_abort(1);
-    }
-    strcpy(config.var_name, token);
+void parse_filter_string(const std::string &filter, ReportConfiguration &config) {
+    std::istringstream iss(filter);
+    std::string token;
+    std::getline(iss, config.mech_name, '.');
+    std::getline(iss, config.var_name, '.');
 }
 
-std::vector<ReportConfiguration> create_report_configurations(const char* conf_file,
-                                                              const char* output_dir,
-                                                              std::string& spikes_population_name) {
+std::vector<ReportConfiguration> create_report_configurations(const std::string &conf_file,
+                                                              const std::string &output_dir,
+                                                              std::string &spikes_population_name) {
     std::vector<ReportConfiguration> reports;
-    int num_reports = 0;
-    char report_on[REPORT_MAX_NAME_LEN] = "";
-    char raw_line[REPORT_MAX_FILEPATH_LEN] = "";
-    char spikes_population[REPORT_MAX_NAME_LEN] = "";
-    TargetType target_type;
-    int* gids;
+    std::string report_on;
+    int target_type;
```

Comment:
`TargetType` is an `int` internally, why the change? Just curious, did it fail to compile or something?

(Clicked the wrong place, excuse me for re-adding the comment, hehe)

Reply:
It didnt compile since istringstream >> didnt like the enum.. so i read the int and convert it after